ivy/func_wrapper.py

```diff
@@ -1508,6 +1509,53 @@ def _handle_complex_input(
     _handle_complex_input.handle_complex_input = True
     return _handle_complex_input
 
+  
+def handle_backend_invalid(fn: Callable) -> Callable:
+    @functools.wraps(fn)
+    def _handle_backend_invalid(*args, **kwargs):
+        """
+        Check if any of the arguments (or nested arguments) passed to the function are
+        instances of ivy.Array or ivy.NativeArray. If so, it returns the function. If
+        not, it raises an InvalidBackendException.
+
+        Parameters
+        ----------
+        args
+            The arguments to be passed to the function.
+
+        kwargs
+            The keyword arguments to be passed to the function.
+
+        Returns
+        -------
+            The return of the function if the current
+            backend matches the argument backend.
+            If not, it raises an InvalidBackendException
+        """
+        array_indices = ivy.nested_argwhere(
+            [args, kwargs], lambda x: isinstance(x, ivy.Array)
+        )
+        array_vals = ivy.multi_index_nest([args, kwargs], array_indices)
+
+        def func(x):
+            target_backend = ivy.utils.backend.handler._determine_backend_from_args(x)
+            if target_backend is not None and ivy.current_backend() != target_backend:
```

Comment:
let's replace the condition here by `if target_backend is not None and ivy.current_backend_str() != target_backend.backend:` as it's failing all the tests in the CI given that the backend objects are different instances even if the backend is the same. Thanks @mosesdaudu001 😄